core/embed/rust/src/ui/model_tr/component/progress.rs

```diff
@@ -41,24 +42,66 @@ where
 {
     const AREA: Rect = constant::screen();
 
-    pub fn new(
-        title: T,
-        indeterminate: bool,
-        description: T,
-        update_description: fn(&str) -> Result<T, Error>,
-    ) -> Self {
+    pub fn new(indeterminate: bool, description: T) -> Self {
         Self {
-            title: Child::new(Label::centered(title, theme::TEXT_BOLD)),
+            title: None,
             value: 0,
             loader_y_offset: 0,
             indeterminate,
             description: Child::new(Paragraphs::new(
                 Paragraph::new(&theme::TEXT_NORMAL, description).centered(),
             )),
             description_pad: Pad::with_background(theme::BG),
-            update_description,
+            update_description: None,
+            icon: theme::ICON_TICK_FAT,
+        }
+    }
+
+    pub fn with_title(mut self, title: T) -> Self {
+        self.title = Some(Child::new(Label::centered(title, theme::TEXT_BOLD)));
+        self
+    }
+
+    pub fn with_update_description(
+        mut self,
+        update_description: fn(&str) -> Result<T, Error>,
+    ) -> Self {
+        self.update_description = Some(update_description);
+        self
+    }
+
+    pub fn with_icon(mut self, icon: Icon) -> Self {
+        self.icon = icon;
+        self
+    }
+
+    pub fn request_paint(&self, ctx: &mut EventCtx) {
+        if !animation_disabled() {
+            ctx.request_paint();
         }
     }
+
+    pub fn request_paint_all_components(&mut self, ctx: &mut EventCtx) {
+        if !animation_disabled() {
+            ctx.request_paint();
+            if let Some(title) = &mut self.title {
+                title.mutate(ctx, |ctx, _title| {
+                    ctx.request_paint();
+                });
+            }
+            self.description.mutate(ctx, |ctx, _description| {
+                ctx.request_paint();
+            });
+        }
```

Comment:
Does this not work?
```suggestion
        if !animation_disabled() {
            self.request_complete_repaint();
        }
```

Reply:
Unfortunately not, it does not show the description on the second pass (and any future ones) - 
![image](https://github.com/trezor/trezor-firmware/assets/42543243/19d42840-af38-4a69-9315-57ca5decbb53)

Therefore I am manually updating both the title and description

Reply:
I think the problem here is that `Progress::event()`does not pass events down to its children, if you fix that `request_complete_repaint()` should work.

This sounds like something that should be documented, feel free to extend the `trait Component` docstring with a note that passing events to child component is required even if they never return a message :pray:, if it works.

Reply:
Right, calling event on `description` and `title` solved the issue - thanks for troubleshooting.

Done in [19d0344](https://github.com/trezor/trezor-firmware/pull/3440/commits/19d0344b440d01395bd6a7268a591b8d2820caf4)